### PR TITLE
Adds back retry_attempts to botocore

### PR DESCRIPTION
**Pull request by thomasdesr**
# Description

The `retry_attempts` field in `instrumentation-botocore` previously reported how many retries it took botocore to complete an API request (e.g. if the client is rate limited).

It was lost during this refactor https://github.com/open-telemetry/opentelemetry-python-contrib/pull/150 & and I can't find any comments or issues saying it was intentionally dropped (sorry if I missed this!).

So this PR adds it back :D

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- Manual test: https://gist.github.com/thomasdesr/08849445953715f38cf6d29d17b00adf
- Updated the existing botocore automated tests

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/master/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [x] Unit tests have been added
- [ ] Documentation has been updated

(Lmk if you want me to add this to the changelog)

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/__init__.py [hunk 1]
```diff
@@ -153,6 +153,11 @@ def _patched_api_call(self, original_func, instance, args, kwargs):
                             "aws.request_id", req_id,
                         )
 
+                    if "RetryAttempts" in metadata:
+                        span.set_attribute(
+                            "aws.retry_attempts", metadata["RetryAttempts"],
```
Comment (toumorokoshi) [inline instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/__init__.py:158]: Is `aws.retry_attempts` the right choice? The [original code](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/150/files#diff-05ef2bbeb79bb2d595bf8a3f7fa2d475d0f34f80804e5e1ba0d9a253d2c0e0e3L154) seems to just have retry_attempts.

That makes sense to me, as retry attempts, albeit behavior in AWS's boto, is not really in aws-specific construct.

retry attempts actually seems quite generic, it probably should be a semantic convention in the spec.

**Review by toumorokoshi — CHANGES_REQUESTED** [review]
for now, I'd leave out "aws.retry_attempts", because it's not specific to AWS.

**Comment by thomasdesr** [comment]
Thanks for taking a look! Moved back to just plain `retry_attempts` :D

**Review by toumorokoshi — APPROVED** [review]
Thanks! LGTM.

**Review by codeboten — CHANGES_REQUESTED** [review]
Thanks for the PR! It looks great, if you could add an entry to the changelog for this, I can approve and merge.

**Review by codeboten — APPROVED** [review]
Thanks!

**Comment by thomasdesr** [comment]
Thanks for the reviews & project in general :D

**Pull request merged**